language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/VariableAssignmentCodeAction.java

```diff
@@ -223,16 +223,44 @@ private static String getDiagnosedContent(Diagnostic diagnostic, LSContext conte
     }
 
     private static Position offsetInvocation(String diagnosedContent, Position position) {
+//        Need to capture the correct function invocation position in chain & nested invocations
+//        eg. General Invocations: lorry.get_color()
+//            Chain invocations: lorry.get_color().print(10),
+//            Package Prefixes: http:lorry.get_color()
+//            Action invocations: http:lorry->action()
+//            Nested invocations: crypto:hashMd5(str.toBytes())
+//            Field accesses: http:lorry.get_color
+        String content = diagnosedContent;
+        int pendingLParenthesis = 0;
+        boolean loop = true;
+        int count = 0;
+        while (loop) {
+            if (content.length() == 2) {
+                break;
+            }
+            char tailChar = content.charAt(content.length() - 1);
+            char tailPrevChar = content.charAt(content.length() - 2);
+            if (pendingLParenthesis <= 0) {
+                if (tailChar == '.' || tailChar == ':') {
+                    count++;
+                    break;
+                } else if ((tailPrevChar == '-' && tailChar == '>')) {
+                    count += 2;
+                    break;
+                }
+            }
+            if (tailChar == '(') {
+                pendingLParenthesis--;
+            } else if (tailChar == ')') {
+                pendingLParenthesis++;
+            }
+            content = content.substring(0, content.length() - 1);
+            count++;
+        }
+
```

Comment:
Shall we update the logic to handle(ignore) characters inside string literals? :)

Reply:
Yes 👍 , Need to ignore chars inside strings for the calculation.

Reply:
@rasika Maybe it'd be better to add a test case including string literals, if possible :)

Reply:
Yes, added a test case as well.